### PR TITLE
pipewire: update to 0.3.40

### PR DESCRIPTION
**Pull request by WhiredPlanck**
Topic Description
-----------------

Update `pipewire` to 0.3.40

Package(s) Affected
-------------------

`pipewire`

Security Update?
----------------

No

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

File: extra-libs/libcamera/autobuild/beyond [hunk 1]
```diff
@@ -0,0 +1,2 @@
+abinfo 'Fixing the mode of signed IPA modules ...'
+find "$PKGDIR"/usr/lib/libcamera -type f -exec chmod 755 {} +
```
Comment (MingcongBai) [inline extra-libs/libcamera/autobuild/beyond]: Add verbosity here. Also your abinfo doesn't mean much beyond that "I'm doing something here." Say exactly what you did here.

File: extra-libs/libcamera/autobuild/defines [hunk 2]
```diff
@@ -0,0 +1,12 @@
+PKGNAME=libcamera
+PKGSEC=libs
+PKGDES="A cross-platform complex camera support library"
+BUILDDEP="boost doxygen gnutls gstreamer-1-0 gtest jinja2 libevent libtiff \
+          libdwarf lttng-ust meson ninja openssl pyyaml qt-5 sphinx systemd"
+
+# qcam and cam will crash with LTO
```
Comment (MingcongBai) [inline extra-libs/libcamera/autobuild/defines]: Could you add a URL for the upstream report?

File: extra-libs/libfreeaptx/autobuild/defines [hunk 3]
```diff
@@ -0,0 +1,4 @@
+PKGNAME=libfreeaptx
+PKGSEC=sound
+PKGDES="Free implementation of Audio Processing Technology codec (aptX), forked from libopenaptx 0.2.0"
```
Comment (MingcongBai) [inline extra-libs/libfreeaptx/autobuild/defines]: Drop the part about fork, not useful for users.

File: extra-libs/libcamera/autobuild/defines [hunk 4]
```diff
@@ -0,0 +1,12 @@
+PKGNAME=libcamera
+PKGSEC=libs
+PKGDES="A cross-platform complex camera support library"
```
Comment (MingcongBai) [inline extra-libs/libcamera/autobuild/defines]: What do you mean by "complex"?

File: extra-libs/libfreeaptx/autobuild/patches/0001-make-the-makefile-compatible-for-autobuild3.patch [hunk 5]
```diff
@@ -0,0 +1,40 @@
+diff -Naur libfreeaptx-0.1.1/Makefile libfreeaptx-0.1.1.aosc/Makefile
```
Comment (MingcongBai) [inline extra-libs/libfreeaptx/autobuild/patches/0001-make-the-makefile-compatible-for-autobuild3.patch]: `compatible-with` *

Comment (MingcongBai) [reply extra-libs/libfreeaptx/autobuild/patches/0001-make-the-makefile-compatible-for-autobuild3.patch]: Drop `the` as well.

File: extra-libs/libfreeaptx/autobuild/patches/0002-allow-to-override-build-flags.patch [hunk 6]
```diff
@@ -0,0 +1,16 @@
+diff -Naur libfreeaptx-0.1.1/Makefile libfreeaptx-0.1.1.aosc/Makefile
```
Comment (MingcongBai) [inline extra-libs/libfreeaptx/autobuild/patches/0002-allow-to-override-build-flags.patch]: `allow-overriding` *

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-libs/libfreeaptx/autobuild/patches/0001-make-the-makefile-compatible-for-autobuild3.patch (hunk 5 above)
Comment (WhiredPlanck) [reply extra-libs/libfreeaptx/autobuild/patches/0001-make-the-makefile-compatible-for-autobuild3.patch]: OK. I copied the file name from the `libopenaptx` corresponding patch. :P

File: extra-libs/libfreeaptx/autobuild/patches/0002-allow-to-override-build-flags.patch (hunk 6 above)
Comment (WhiredPlanck) [reply extra-libs/libfreeaptx/autobuild/patches/0002-allow-to-override-build-flags.patch]: Sorry for my poor English. :joy:

**Comment by WhiredPlanck** [comment]
LGTM on AMD64

图片

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**